streams/test-utils/src/main/java/org/apache/kafka/streams/test/TestRecord.java

```diff
@@ -0,0 +1,239 @@
+     *                  the timestamp is assigned using Instant.now() or internally tracked time.
+     */
+    public TestRecord(final K key, final V value, final Instant recordTime) {
+        this(key, value, null, recordTime);
+    }
+
+    /**
+     * Creates a record
+     *
+     * @param key The key that will be included in the record
+     * @param value The record contents
+     * @param headers The headers that will be included in the record
+     */
+    public TestRecord(final K key, final V value, final Headers headers) {
+        this.key = key;
+        this.value = value;
+        this.headers = new RecordHeaders(headers);
+        this.recordTime = null;
+    }
+    
+    /**
+     * Creates a record
+     *
+     * @param key The key that will be included in the record
+     * @param value The record contents
+     */
+    public TestRecord(final K key, final V value) {
+        this.key = key;
+        this.value = value;
+        this.headers = new RecordHeaders();
+        this.recordTime = null;
+    }
+
+    /**
+     * Create a record with no key
+     *
+     * @param value The record contents
+     */
+    public TestRecord(final V value) {
+        this(null, value);
+    }
+
+    /**
+     * Create a TestRecord from ConsumerRecord
+     *
+     * @param record The record contents
+     */
+    public TestRecord(final ConsumerRecord<K, V> record) {
+        this(record.key(), record.value(), record.headers(), record.timestamp());
+    }
+
+    /**
+     * Create a TestRecord from ProducerRecord
+     *
+     * @param record The record contents
+     */
+    public TestRecord(final ProducerRecord<K, V> record) {
+        this(record.key(), record.value(), record.headers(), record.timestamp());
+    }
+
+    /**
+     * @return The headers
+     */
+    public Headers headers() {
+        return headers;
+    }
+
+    /**
+     * @return The key (or null if no key is specified)
+     */
+    public K key() {
+        return key;
+    }
+
+
+    /**
+     * @return The value
+     */
+    public V value() {
+        return value;
+    }
+
+    /**
+     * @return The timestamp, which is in milliseconds since epoch.
+     */
+    public Long timestamp() {
+        return this.recordTime == null ? null : this.recordTime.toEpochMilli();
+    }
+
+    /**
+     * @return The headers
+     */
+    public Headers getHeaders() {
+        return headers;
+    }
+
+    /**
+     * @return The key (or null if no key is specified)
+     */
+    public K getKey() {
+        return key;
+    }
+
+    /**
+     * @return The value
+     */
+    public V getValue() {
+        return value;
+    }
+
+    /**
+     * @return The recordTime as Instant
+     */
+    public Instant getRecordTime() {
+        return recordTime;
+    }
+
+    @Override
+    public String toString() {
+        final String headers = this.headers == null ? "null" : this.headers.toString();
```

Comment:
You should double-check, but I think Java will automatically do this in string concatenation. (It was news to me when I learned it a couple of months back)

Reply:
Replaced with another style toString